packages/flutter_tools/lib/src/base/error_handling_io.dart

```diff
@@ -762,6 +766,19 @@ void _handlePosixException(Exception e, String? message, int errorCode, String?
   _throwFileSystemException(errorMessage);
 }
 
+void _handleMacOSException(Exception e, String? message, int errorCode, String? posixPermissionSuggestion) {
+  // https://github.com/apple/darwin-xnu/blob/master/bsd/dev/dtrace/scripts/errno.d
+  const int ebadarch = 86;
+  if (errorCode == ebadarch) {
+    final StringBuffer errorBuffer = StringBuffer();
+    errorBuffer.writeln(message);
+    errorBuffer.writeln('Flutter requires the Rosetta translation environment on ARM Macs. Try running:');
+    errorBuffer.writeln('  sudo softwareupdate --install-rosetta --agree-to-license');
+    _throwFileSystemException(errorBuffer.toString());
```

Comment:
should we tool exit here?

Reply:
That's what `_throwFileSystemException` does.
https://github.com/flutter/flutter/blob/8b32ac7a512b35b2b01da949ff41be814d025207/packages/flutter_tools/lib/src/base/error_handling_io.dart#L814-L822